src/plugins/intel_gpu/src/graph/include/program_node.h

```diff
@@ -175,6 +180,20 @@ struct program_node {
         return layouts;
     }
 
+    layout get_input_layout(size_t idx = 0) const {
+       return get_dependency(idx).get_output_layout();
+    }
```

Comment:
Is it expected to implicitly invalidate all users' layouts in some cases? Shouldn't we use safer `invalidate_users_if_changed=false` as a default? As I can see, there are some places where
`node.get_dependency(idx).get_output_layout(false)` is used